plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -434,15 +434,11 @@ public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTabl
 
     private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName tableName)
     {
-        IcebergTableName name = IcebergTableName.from(tableName.getTableName());
-        if (name.getTableType() == DATA) {
-            return Optional.empty();
-        }
-
+        String name = IcebergTableName.tableNameFrom(tableName.getTableName());
```

Comment:
The table handle should be retrieved only when dealing with non-DATA system tables.